docs/usage/caching.rst

```diff
@@ -15,59 +15,63 @@ expected. Litestar comes with a simple mechanism for caching:
 
-
-.. note::
-    If the default ``default_expiration`` is set to ``None``, setting up the route handler with ``cache=True`` will keep
-    the response in cache indefinitely.
+.. note:: If the default :paramref:`~litestar.config.response_cache.ResponseCacheConfig.default_expiration` is set to
+    ``None``, setting up the route handler with :paramref:`~litestar.handlers.HTTPRouteHandler.cache` set to ``True``
+    will keep the response in cache indefinitely.
 
 Alternatively you can specify the number of seconds to cache the responses from the given handler like so:
 
 .. code-block:: python
+    :caption: Caching the response for 120 seconds by setting the :paramref:`~litestar.handlers.HTTPRouteHandler.cache`
+      parameter to the number of seconds to cache the response.
+    :emphasize-lines: 4
 
-   from litestar import get
+    from litestar import get
 
 
-   @get("/cached-path", cache=120)  # seconds
-   def my_cached_handler() -> str: ...
+    @get("/cached-path", cache=120)  # seconds
+    def my_cached_handler() -> str: ...
 
 
-If you want the response to be cached indefinitely, you can pass the :class:`.config.response_cache.CACHE_FOREVER`
+If you want the response to be cached indefinitely, you can pass the :class:`~.config.response_cache.CACHE_FOREVER`
 sentinel instead:
 
 .. code-block:: python
+    :caption: Caching the response indefinitely by setting the :paramref:`~litestar.handlers.HTTPRouteHandler.cache`
+      parameter to :class:`~litestar.config.response_cache.CACHE_FOREVER`.
 
-   from litestar import get
-   from litestar.config.response_cache import CACHE_FOREVER
+    from litestar import get
+    from litestar.config.response_cache import CACHE_FOREVER
 
 
-   @get("/cached-path", cache=CACHE_FOREVER)  # seconds
-   def my_cached_handler() -> str: ...
-
+    @get("/cached-path", cache=CACHE_FOREVER)
+    def my_cached_handler() -> str: ...
 
 Configuration
 -------------
 
 You can configure caching behaviour on the application level by passing an instance of
-:class:`ResponseCacheConfig <.config.response_cache.ResponseCacheConfig>` to the :class:`Litestar instance <.app.Litestar>`.
-
+:class:`~.config.response_cache.ResponseCacheConfig` to the :class:`Litestar instance <.app.Litestar>`.
 
 Changing where data is stored
 +++++++++++++++++++++++++++++
 
-By default, caching will use the :class:`MemoryStore <.stores.memory.MemoryStore>`, but it can be configured with
-any :class:`Store <.stores.base.Store>`, for example :class:`RedisStore <.stores.redis.RedisStore>`:
+By default, caching will use the :class:`~.stores.memory.MemoryStore`, but it can be configured with
+any :class:`~.stores.base.Store`, for example :class:`~.stores.redis.RedisStore`:
 
 .. code-block:: python
+    :caption: Using Redis as the cache store.
 
-   from litestar.config.cache import ResponseCacheConfig
-   from litestar.stores.redis import RedisStore
+    from litestar.config.cache import ResponseCacheConfig
```

Comment:
I think this is a remnant of an old pattern. IIUC the snippet in the docs would no longer work.

The following is an MCVE that shows how to use a custom store using the current interfaces.

```py
import asyncio
from litestar import Litestar, get
from litestar.config.response_cache import ResponseCacheConfig
from litestar.stores.redis import RedisStore


@get(cache=10)
async def something() -> str:
    await asyncio.sleep(1)
    return "something"


redis_store = RedisStore.with_client(url="redis://localhost", port=6379, db=0)
cache_config = ResponseCacheConfig(store="redis_backed_store")
app = Litestar(
    [something],
    stores={"redis_backed_store": redis_store},
    response_cache_config=cache_config,
)
```

Some may argue this is not good DX, I personally do not find this any difficult.

Reply:
This is more evidence towards us needing to move any examples into `examples/` and have them tested with each PR so they dont go stale.

We dont have to show the whole app, we could `:linenos:` it, but we should really work on that